### ISSUE TITLE
Frontier Gate Boost+ trial jit bug

### ISSUE THREAD

**Issue by daniel229** [hrydgard/ppsspp#7066]
It hanged after completing certain quests,but if save and load savestate before,it won't hang.The retail version also has the bug.

the error messages
https://gist.github.com/daniel229/72f54bdf2ae9e04a091e

Frontier Gate Boost+ trial download
http://dl.frontiergate.konamionline.com/frontiergate/FGBP_trial.zip

savestate(rename jpg to rar)
![npjh90323_1 00_0](https://cloud.githubusercontent.com/assets/3481559/4963300/43749e0a-6707-11e4-90a7-278246747dcd.jpg)

A video show how to accept quest and complete the quest.
http://youtu.be/KZ52N713nKY

But the jit bug looks like it happened during the quest list,maybe it not necessary to complete the quest.
save and load savestate in the following picture,it still get red error codes and hang after completing the quest.
![01](https://cloud.githubusercontent.com/assets/3481559/4963305/f4c6bbc0-6707-11e4-86cc-32b89e5ce866.png)

save and load savestate in the following picture,it get invalid addresses and hang after completing the quest.
![02](https://cloud.githubusercontent.com/assets/3481559/4963308/3a338076-6708-11e4-851e-51e0fa6dc9fe.png)

save and load savestate after accept the quest and quit the quest list like the following picture,it won't get errors,and not hang after completing the quest.
![03](https://cloud.githubusercontent.com/assets/3481559/4963314/8d3a2194-6708-11e4-81a5-6846f2e7136c.png)

**Comment by sum2012** [comment]
baidu user report that the bug still exist

**Comment by unknownbrackets** [comment]
I'm not able to reproduce this now.  I'm not sure what change fixed it, or if I'm just doing it wrong.

Does this still reproduce?

-[Unknown]

**Comment by unknownbrackets** [comment]
Okay, I tried this again just now and can't reproduce it.  Here's what I tried:
* I started the game, loaded in-game save data, and did the mission in question (run straight through, off to the left, and talk to the woman.)  Then returned to base.
* I restarted the game, saved state at the mission menu, completed the mission, loaded the save state and completed the mission again.  Then returned to base.
* I restarted PPSSPP, loaded the save state, and completed the mission.  Then returned to base.
* I restarted PPSSPP, and loaded the game from a save.  At the mission select, I saved state and then immediately loaded the state.  Then I completed the mission and returned to base.

I did all of these using jit and tried some of them with interpreter or IR interpreter (note: with #16396.)  They all worked fine.  I think this has been fixed, so I'm going to close it.

-[Unknown]
